woocommerce/woocommerce-ios
# [Payments Menu] Add Pay in Person toggle

Closes: #11101 


## Description


As part of the SwiftUI rewrite of the Payments Menu, this PR makes minor changes to the existing Pay in Person toggle VM, and reuses it with a new SwiftUI view.

Functionality is the same as the existing view.

### Known issues
- Because we don't have background onboarding yet, the Learn More link doesn't always have the URL for the correct payment gateway extension (Stripe/WooPayments.) This will be addressed in #11104.

## Testing instructions


1. Launch the app from Xcode
2. Navigate to `Menu > Settings > Experimental Features`
3. Toggle `SwiftUI Payments Menu`
4. Navigate to `Menu > Payments`
5. Observe that the Pay in Person toggle is shown, and has the current value for whether Cash on Delivery is enabled for the store.
6. Tap `Learn More` – observe that a web view sheet is opened for (hopefully!) the correct extension.
7. Observe that the `payments_hub_cash_on_delivery_toggle_learn_more_tapped` analytic event is tracked
8. Close the web view and toggle the setting; observe that it makes a network request and updates the setting for your site (verify in WP-Admin > Payments > Settings > Other Payment Gateways)
9. `payments_hub_cash_on_delivery_toggled` and `disable_cash_on_delivery_success` or `enable_cash_on_delivery_success` are tracked.

### Failure testing

1. Set up Proxyman with a breakpoint on POST requests to `https://public-api.wordpress.com/rest/v1.1/jetpack-blogs/*/rest-api/`
2. Toggle the Pay in Person option
3. Abort at the Proxyman breakpoint
4. Observe that a notice is shown
5. Check that `enable_cash_on_delivery_failed` or `disable_cash_on_delivery_failed` were tracked, as appropriate.
6. Tap Retry
7. Execute the Proxyman breakpoint
8. Observe that the change is completed.
9. 

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/2472348/4bd4d35a-f324-4dbf-8e07-6ead7bdfef71

### Failure testing

https://github.com/woocommerce/woocommerce-ios/assets/2472348/6fd52ea0-4d1c-4660-9da2-da423a74e173


---
